client/me/purchases/purchase-item/index.jsx

```diff
@@ -93,21 +97,21 @@ class PurchaseItem extends Component {
 			}
 
 			return translate( 'Expires on %s', {
-				args: purchase.expiryMoment.format( 'LL' ),
+				args: expiry && expiry.format( 'LL' ),
```

Comment:
I'm not sure if all these `expiryDateValid` and `expiry` checks are needed. They don't even seem to do a good job. If we display a message that says `Expires on false`, then what's the point? And it seems that the app worked just fine without them until now.

This is my main concern with this PR.

Reply:
Yeah, I can see your point. I'll go ahead and remove all of the `*Valid`, except for `renewDateValid`, since that one has special logic in place which I still think belongs in the "data" layer.

Reply:
Done.

Reply:
I think this still stands for the `expiry && ` checks though?

Reply:
Removed!